src/components/Text.tsx

```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+
+export interface TextProps extends React.HTMLAttributes<HTMLParagraphElement> {
+  asChild?: string;
+}
+
+const Text = React.forwardRef<HTMLParagraphElement, TextProps>(
+  ({ className, asChild = "p", ...props }, ref) => {
+    const Comp = asChild === "p" ? "p" : "h3";
```

Comment:
WDYT?

cc @DaniloArantesF

Reply:
Made a few changes to allow other element types and added base styles to each variant